### PR TITLE
Move profile flag to common flags

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

Move profile flag to common flags for server/agent so that it can be loaded from config YAML. Also simplified flag use a bit to get rid of unnecessary iteration over flag lists.

#### Types of Changes ####

* CLI

#### Verification ####

* Start rke2 with `profile: cis-1.5`in config.yaml
* Note that user and sysctls are validated; no "Not running in CIS mode" warning shown.

#### Linked Issues ####

Related to #351

### CODE REVIEW

File: pkg/rke2/rke2.go [hunk 1]
```diff
@@ -53,27 +52,8 @@ func Agent(clx *cli.Context, cfg Config) error {
...
-		case cli.StringFlag:
-			if t.Name == "profile" && t.Destination != nil && *t.Destination != "" {
-				cisMode = true
-			}
-		default:
-			// nothing to do. Keep moving.
-		}
-	}
+	cisMode = clx.String("profile") != ""
+	dataDir := clx.String("data-dir")
```
Comment (cjellick) [inline pkg/rke2/rke2.go:56]: I have to ask why? 
Why did we iterate through the flags before and why do we not need to now.

File: pkg/cli/cmds/agent.go [hunk 2]
```diff
@@ -54,8 +53,6 @@ func NewAgentCommand() cli.Command {
 }
 
 func AgentRun(clx *cli.Context) error {
-	if profile == "" {
-		logrus.Warn("not running in CIS 1.5 mode")
-	}
+	validateCISreqs()
```
Comment (cjellick) [inline pkg/cli/cmds/agent.go:56]: Teasing out the logic of the validateCISreqs, I now wonder if we will refuse to start a worker node because the etcd user doesn't exist even though that user need not exist on a worker node.

Comment (cjellick) [reply pkg/cli/cmds/agent.go:56]: I think that bug may already exist, but this change here made me think about it.

Comment (cjellick) [reply pkg/cli/cmds/agent.go:56]: The answer is: yes it already exists.

File: pkg/rke2/rke2.go (hunk 1 above)
Comment (brandond) [reply pkg/rke2/rke2.go:56]: I suspect this was either necessary when using a different CLI library, or sometime didn't know the right way to do it.

File: pkg/cli/cmds/agent.go [hunk 3]
```diff
@@ -54,8 +54,16 @@ func NewAgentCommand() cli.Command {
 }
 
 func AgentRun(clx *cli.Context) error {
-	if profile == "" {
+	switch profile {
+	case "cis-1.5":
+		if err := validateCISreqs(); err != nil {
```
Comment (cjellick) [inline pkg/cli/cmds/agent.go:59]: why did you need to move the validateCISreqs() logic?

Part of the reason I ask is that I think you are oging to have to change it to fix the etcd user bug i mentioned

**Review by cjellick — APPROVED** [review]

File: pkg/cli/cmds/agent.go (hunk 3 above)
Comment (cjellick) [reply pkg/cli/cmds/agent.go:59]: answered offline and the TL;DR is that this needed moved once the arg was moved out to the server and agent commands

File: pkg/cli/cmds/agent.go [hunk 4]
```diff
@@ -54,8 +54,16 @@ func NewAgentCommand() cli.Command {
 }
 
 func AgentRun(clx *cli.Context) error {
-	if profile == "" {
+	switch profile {
```
Comment (briandowns) [inline pkg/cli/cmds/agent.go:57]: This would probably make sense to have in a function in root.go that takes the profile string as an arg and does this logic so it's not needed to be maintained in 2 different places. More easily extensible too if/when we move to 1.6, etc. Not blocking however it'd be cool to have an issue that captures this.

**Review by briandowns — APPROVED** [review]

File: pkg/cli/cmds/agent.go (hunk 4 above)
Comment (brandond) [reply pkg/cli/cmds/agent.go:57]: I agree 100%. I had originally moved this into the validateCISreqs function, but Craig asked me to reduce the amount of change so I went for a more minimal approach at the cost of some repetition.

Comment (cjellick) [reply pkg/cli/cmds/agent.go:57]: +1. He basically had that, but I asked him to minimize the pr to the smallest change possible for the purpose of reviewing and merging as quickly as possible.

Comment (briandowns) [reply pkg/cli/cmds/agent.go:57]: Cool. Do we have an issue for this for after GA?
